extensions/openstack/openstack-impl/src/main/java/org/mqnaas/extensions/openstack/capabilities/impl/OpenstackRootResourceProvider.java

```diff
@@ -108,15 +118,18 @@ public void activate() throws ApplicationActivationException {
 		log.info("Initializing OpenstackRootResourceProvider capability for resource " + resource.getId());
 
 		try {
-			novaClient = jcloudsClientProvider.getClient(resource);
+			novaClient = clientProviderFactory.getClientProvider(IJCloudsNovaClientProvider.class).getClient(resource);
+			neutronClient = clientProviderFactory.getClientProvider(IJCloudsNeutronClientProvider.class).getClient(resource);
 		} catch (EndpointNotFoundException e) {
```

Comment:
Initializing clients to be reused? I think it will probably be better to call getClient() in each call.